src/gurt/tests/test_gurt_telem_consumer.c

```diff
@@ -37,6 +37,16 @@ init_tests(void **state)
 	return d_log_init();
 }
 
+static void
+test_shmem_removed(void **state)
+{
+	uint64_t	*shmem;
+	int		simulated_srv_idx = 100;
+
+	shmem = d_tm_get_shared_memory(simulated_srv_idx);
+	assert_null(shmem);
+}
```

Comment:
I'm probably just missing something, but I'm curious how one guarantees this test and the producer work in lockstep, so that we get the timing right.

Reply:
It's only based on the order in which the test programs are specified to run in src/gurt/tests/Sconscript.  Currently, the producer test runs before the consumer test.  If someone changes that order, then the test won't succeed.  But, anyone who changes that in the future will see a test failure and will see their mistake.